### PR TITLE
Fix for alphabetical recursive call issue

### PR DESCRIPTION
**Pull request by SiriusStarr**
This adds a test case for the bug causing #5

### CODE REVIEW

File: tests/CognitiveComplexityTest.elm [hunk 1]
```diff
@@ -611,6 +611,41 @@ fun5 n =
                             "fun5": 1
                           }
                        }"""
+        , test "recursive call complexity should not depend on alphabetical order" <|
+            \() ->
+                """module A exposing (..)
+
+b = b
+
```
Comment (jfmengels) [inline tests/CognitiveComplexityTest.elm:619]: I don't think this compiles, so I don't know how good of a test case this is 🤔

Comment (SiriusStarr) [reply tests/CognitiveComplexityTest.elm:620]: Now it does.

Comment (jfmengels) [reply tests/CognitiveComplexityTest.elm:620]: 👍

**Comment by SiriusStarr** [comment]
As functions were visited in alphabetical order, they would be marked
`Done`, leading to recursion not getting checked if the call was to an
"earlier" function.

Fixes #5

**Comment by SiriusStarr** [comment]
(Incidentally, you should probably bump the `elm-review` dependency before this or anything else gets released, given bugfixes, since it's still at `2.10.0`.)

**Comment by jfmengels** [comment]
Awesome work @SiriusStarr! Thank you for applying your great :female_detective: skills on this!

**Pull request merged**